lib/miq_expression/remove_equality_visitor.rb

```diff
@@ -0,0 +1,86 @@
+class MiqExpression
+  class RemoveEqualityCollector
+    def initialize(node_to_remove, binds)
+      @node_to_remove = node_to_remove
+      @binds = binds
+    end
+
+    attr_reader :node_to_remove
+
+    def next_bind_value
+      binds.shift.value_for_database
+    end
+  end
+  # rubocop:disable Naming/MethodName
+  class RemoveEqualityVisitor < Arel::Visitors::Reduce
```

Comment:
I don't think this exists in Rails 6... possibly in Rails 5.2 depending on what version of `ActiveRecord` we bumped to in the upgrade:

https://github.com/rails/arel/pull/512

Reply:
That said, it looks like it was effectively `Arel::Visitors::Visitor` anyway.

Reply:
we can cross that bridge when we upgrade to 6.0

Reply:
This is a "I did a `bundle open activerecord` and couldn't find this class", not a "this might be an issue in the future" thing.  I see this being a problem right now, not later.

Reply:
https://github.com/rails/rails/blob/5-2-stable/activerecord/activerecord.gemspec#L34

https://github.com/rails/arel/blob/v9.0.0/lib/arel/visitors/reduce.rb

5-1 uses v8 ==> https://github.com/rails/arel/blob/v8.0.0/lib/arel/visitors/reduce.rb

I think it is good for rails 5.1 and 5.2.
rails 6.0 just uses visitor